balancer/balancer.go

```diff
@@ -410,6 +424,15 @@ type SubConnState struct {
 	// ConnectionError is set if the ConnectivityState is TransientFailure,
 	// describing the reason the SubConn failed.  Otherwise, it is nil.
 	ConnectionError error
+	// connectedAddr contains the connected address when ConnectivityState is
+	// Ready. Otherwise, it is indeterminate.
+	connectedAddress resolver.Address
+}
+
+func (lhs SubConnState) Equal(rhs SubConnState) bool {
```

Comment:
Will we delete this entirely when `connectedAddress` is removed?  If so, ignore.  If not, then should this accept pointers instead?

Reply:
Done.

Reply:
Removed entirely.

This was only added to get [this single check in `xds/internal/balancer/outlierdetection/balancer_test.go`](https://github.com/grpc/grpc-go/blob/45d44a736ec6cdcb73f9411bf6a1c2d1abea1956/xds/internal/balancer/outlierdetection/balancer_test.go#L854-L859) to pass, but that doesn't work if it uses pointers. I've removed this in favor of adding `balancer.SubConnState{}` to the `cmp.AllowUnexported`.